core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -164,6 +172,26 @@ public void start(final Listener<RespT> observer, Metadata headers) {
     checkNotNull(observer, "observer");
     checkNotNull(headers, "headers");
 
+    // Create the context
+
+    // Same time used to compute all timeouts, in order to avoid inaccuracies and negative
+    // timeouts.
+    final long now = System.nanoTime();
+    final Deadline callDeadline = callOptions.getDeadlineNanoTime() != null
+        ? Deadline.after(callOptions.getDeadlineNanoTime() - now, NANOSECONDS)
```

Comment:
Consider implementing Deadline.at(instant, units) to avoid the unnecessary call to nanoTime just to compute the base.